File: 6.data insertion.py
# write a program to insert an element at any specific position of a array.

import numpy as np
arr = np.zeros(50,int)
a = int(input("enter the number of elements of the array: "))
print("enter elements: ")
for i in range(0,a):
	arr[i] = int(input())

ele = int(input("enter element: "))
pos = int(input("enter position to enter: "))
if(pos > a+1):
	print("out of index")

else:
    for i in range(a,pos-2,-1):
        arr[i] = arr[i-1]
        
    arr[pos-1] = ele

    for i in range(0,a+1):
        print(arr[i],end=" ")
	

File: 6.data insertion.c
// write a program to insert an element at any specific position of a array. 

#include<stdio.h>
int main(){
	int arr[100],n,i,pos,elem;
	printf("enter the number of elements of the array: \n");
	scanf("%d",&n);
	
	printf("enter %d elements\n",n);
	
	for(i = 0;i<n;i++){
		scanf("%d",&arr[i]);
		}
	
	printf("enter the element to insert and position: \n");
	scanf("%d %d",&elem,&pos);
	
	for(i = n-1;i>=pos-1;i--){
		arr[i+1] = arr[i];
		}
	
	arr[pos-1] = elem;
	
	printf("new array is: ");
	for(i = 0;i<=n;i++){
		printf("%d ",arr[i]);
		}
	return 0;
}